compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1673,6 +1673,16 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
+        for def_ids in associated_types.values_mut() {
+            for def_id in def_ids.clone() {
```

Comment:
can we use retain instead of cloning here?

Reply:
It's a HashSet, not a Vec. I should take a second look, but I didn't find a good alternative